src/typing_extensions.py

```diff
@@ -3524,16 +3525,41 @@ def _raise_attribute_error(self, name: str) -> Never:
         def __repr__(self) -> str:
             return self.__name__
 
-        def __getitem__(self, parameters):
-            if not isinstance(parameters, tuple):
-                parameters = (parameters,)
-            parameters = [
-                typing._type_check(
-                    item, f'Subscripting {self.__name__} requires a type.'
-                )
-                for item in parameters
-            ]
-            return typing._GenericAlias(self, tuple(parameters))
+        if sys.version_info >= (3, 11):
+            def __getitem__(self, parameters):
+                if not isinstance(parameters, tuple):
+                    parameters = (parameters,)
+                    parameters = [
+                        typing._type_check(
+                            item, f'Subscripting {self.__name__} requires a type.'
+                        )
+                        for item in parameters
+                    ]
+                return typing._GenericAlias(self, tuple(parameters))
+        else:
+            def _check_parameter(self, item, typ=_marker):
+                # Allow [], [int], [int, str], [int, ...], [int, T]
+                if not isinstance(typ, ParamSpec) and typ is not _marker:
+                    return typing._type_check(
+                            item, f'Subscripting {self.__name__} requires a type.'
+                    )
+                if item is ...:
+                    return ...
+                elif isinstance(item, list) and typ is not _marker:
+                    return [self._check_parameter(arg) for arg in item]
+                else:
+                    return typing._type_check(
+                        item, f'Subscripting {self.__name__} requires a type.'
+                    )
+
+            def __getitem__(self, parameters):
+                if not isinstance(parameters, tuple):
+                    parameters = (parameters,)
+                    parameters = [
+                        self._check_parameter(item, typ)
+                        for item, typ in zip(parameters, self.__type_params__)
```

Comment:
What if the two sequences are of different length? Do we test this case?

Reply:
Just thinking aloud:

You think of a case like this?

```
ListOrSetT = TypeAliasType("ListOrSetT", Union[List[T], Set[T]], type_params=(T,))
WrongSubscripted = ListOrSetT[int, bool]
```

Reply:
Those are useful (they should throw errors I believe). But in the case where there are multiple parameters and they match, we need to make sure `_check_parameter` is called.

Reply:
I am not sure about python 3.13. At least in Python 3.12.5 both cases pass without errors using `typing.TypeAliasType`, and for both cases produce no error. Would you suggest making it stricter? Imo the Python 3.12 variant seems to be very lenient.

Reply:
Ah sorry, I should have checked what I did in CPython. We should match the CPython behavior as precisely as possible.

Reply:
I've now handled this case and added test cases.